src/browser_action/on-each-interaction.js

```diff
@@ -11,43 +11,139 @@
+    const group = {
+      startTime: null,
+      processingStart: null,
+      processingEnd: null
     }
 
-    // Will report as a single interaction even if parts are in separate frames.
-    // Consider splitting by animation frame.
-    for (const interaction of Object.values(interactions)) {
-      const entry = interaction.reduce((prev, curr) => prev.duration >= curr.duration ? prev : curr);
-      const value = entry.duration;
-
-      callback({
-        attribution: {
-          eventEntry: entry,
-          eventTime: entry.startTime,
-          eventType: entry.name,
-        },
-        entries: interaction,
-        name: 'Interaction',
-        rating: valueToRating(value),
-        value,
-      });
+    // Go though the list of all events (not just those with interactionIds)
+    // do get details of what was in the longest frame and the earliest and latest event timestamps
+    for (const entry of allEvents) {
+        const renderTime = entry.startTime + entry.duration;
+        // If a previous render time is within 8ms of the largest render time,
+        // assume they were part of the same frame so include them
+        if (Math.abs(renderTime - largestRenderTime) <= 8) {
+          longestFrameEntries.push(entry);
+          group.startTime = group.startTime ? Math.min(entry.startTime, group.startTime) : entry.startTime;
+          group.processingStart = group.processingStart ?
+            Math.min(
+              entry.processingStart,
+              group.processingStart,
+            ) :
+            entry.processingStart;
+          group.processingEnd = group.processingEnd ? Math.max(entry.processingEnd, group.processingEnd) : entry.processingEnd;
+        }
     }
+
+    // For the breakdowns we need to know the first and last interaction events (i.e. those wiht interactionIds) in that longest list
+    const longestFrameInteractionEntries = longestFrameEntries.filter((entry) => entry.interactionId);
+    // Entries should be in order so can use that to find first and last
+    const firstInteractionEntry = longestFrameInteractionEntries[0];
+    const lastInteractionEntry = longestFrameInteractionEntries.slice(-1)[0];
+
+    // Filter further to get the entries entry (the longest interactionId events wiht the longest durations)
+    const longestInteractionEntries = longestFrameInteractionEntries.filter((entry) => entry.interactionId  === longestEntry.interactionId && entry.duration === largestDuration);
+
+    // Sometimes target is not set so look across all events with that interactionId to get it
+    const firstInteractionEntryWithTarget = interactions.find((entry) => longestEntry.interactionId === longestEntry.interactionId && entry.target);
+
+    // Filter down LoAFs to ones that interested any event startTime and any processingEnd
+    // Theb LoAF processing the last script in that frame will then present that frame so
+    // we get rAF and other rendering work too with this intrsection.
```

Comment:
```suggestion
    // we get rAF and other rendering work too with this intersection.
```
Or maybe "interaction"?